app/src/main/java/com/simply/birthdayapp/presentation/ui/components/CleanableSearchBar.kt

```diff
@@ -0,0 +1,100 @@
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.simply.birthdayapp.R
+
+@Composable
+fun CleanableSearchBar(
+    query: String,
+    onQueryChange: (String) -> Unit = {},
+    onSearch: () -> Unit = {},
+) {
+    CompositionLocalProvider(
+        LocalTextSelectionColors provides TextSelectionColors(
+            handleColor = MaterialTheme.colorScheme.tertiary,
+            backgroundColor = MaterialTheme.colorScheme.tertiary,
+        )
+    ) {
+        TextField(
+            modifier = Modifier
+                .padding(horizontal = 20.dp, vertical = 10.dp)
+                .fillMaxWidth(),
+            value = query,
+            onValueChange = { onQueryChange(it) },
+            placeholder = {
+                Text(
+                    text = stringResource(R.string.search),
+                    fontFamily = FontFamily(Font(R.font.karma_medium)),
+                    color = Color.Gray,
+                )
+            },
+            trailingIcon = {
+                if (query.isNotEmpty()) {
+                    IconButton(onClick = { onQueryChange("") }) {
+                        Icon(
+                            imageVector = Icons.Outlined.Clear,
+                            contentDescription = stringResource(R.string.clear),
+                            tint = Color.Gray,
+                        )
+                    }
+                } else {
+                    IconButton(onClick = onSearch) {
+                        Icon(
+                            painter = painterResource(id = R.drawable.ic_search),
+                            contentDescription = stringResource(R.string.search),
+                            tint = Color.Gray,
+                        )
+                    }
+                }
+            },
+            keyboardOptions = KeyboardOptions(
+                keyboardType = KeyboardType.Text,
+                imeAction = ImeAction.Search,
+            ),
+            keyboardActions = KeyboardActions(
+                onSearch = { onSearch() },
+            ),
+            singleLine = true,
+            shape = CircleShape,
+            colors = TextFieldDefaults.colors(
+                focusedContainerColor = MaterialTheme.colorScheme.tertiaryContainer,
+                unfocusedContainerColor = MaterialTheme.colorScheme.tertiaryContainer,
+                focusedIndicatorColor = Color.Transparent,
+                unfocusedIndicatorColor = Color.Transparent,
+                cursorColor = MaterialTheme.colorScheme.tertiary,
+            ),
+        )
+    }
+}
+
+@Preview
+@Composable
+private fun CleanableSearchBar() {
+    CleanableSearchBar(query = "")
```

Comment:
Give it as a default value to the `query` argument to not specify in the preview.

Reply:
[Given](https://github.com/SimplyTechnologies/android-internship/pull/14/commits/2b0a0df2c5d2218c5cac09cbd6953e854debf026)